### PR TITLE
Testcontainer for RabbitMQ

### PR DESCRIPTION
**Pull request by leopetter**
This PR adds a testcontainer for rabbitMQ.
The testcontainer is implemented along side a wrapper for the producer and consumer for rabbitMQ, it is well tested and includes an example like the other testcontainers.

### CODE REVIEW

File: examples/rabbitmq/rabbitmq_example.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          false,  // exclusive
+          false,  // no local
+          false,  // no wait
+          nil,    // args
+     )
+     if err != nil {
+		log.Fatalf("Failed to register a consumer: %v", err)
+	}
+
+     forever := make(chan bool)
```
Comment (romnnn) [inline examples/rabbitmq/rabbitmq_example.go:69]: Nit: Rename to `done`

File: examples/rabbitmq/rabbitmq_example.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+     go func() {
+          for d := range msgs {
+              log.Infof("%s \n", d.Body)
+              if len(d.Body) < 20 {
+                shortMessageCounter++
+              }
+              consumedCount++
+              if consumedCount == publishCount {
+                forever <- true
```
Comment (romnnn) [inline examples/rabbitmq/rabbitmq_example.go:81]: Return here to exit the goroutine

File: rabbitmq/producer.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+func failOnError(err error, msg string) {
+        if err != nil {
+                log.Fatalf("%s: %s", msg, err)
+        }
+}
+
+func Setup(options RabbitMQOptions) (*amqp.Connection, *amqp.Channel){
+  // 1. Create the connection to RabbitMQ
+  url := fmt.Sprintf("amqp://guest:guest@%s:%d", options.Host, options.Port)
```
Comment (romnnn) [inline rabbitmq/producer.go:25]: Extract into function `func (options RabbitMQOptions) ConnectionURI() string` that provides this

File: rabbitmq_test.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+          false,  // exclusive
+          false,  // no local
+          false,  // no wait
+          nil,    // args
+     )
+     if err != nil {
+		t.Fatalf("Failed to register a consumer: %v", err)
+	}
+
+     forever := make(chan bool)
```
Comment (romnnn) [inline rabbitmq_test.go:69]: See example comment

File: rabbitmq_test.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+     consumedCount := 0
+
+     go func() {
+          for range msgs {
+              consumedCount++
+              if consumedCount > 10 {
+                forever <- true
```
Comment (romnnn) [inline rabbitmq_test.go:76]: See example comment

File: rabbitmq/producer.go (hunk 3 above)
Comment (leopetter) [reply rabbitmq/producer.go]: Fixed

File: examples/rabbitmq/rabbitmq_example.go (hunk 1 above)
Comment (leopetter) [reply examples/rabbitmq/rabbitmq_example.go]: Fixed

File: examples/rabbitmq/rabbitmq_example.go (hunk 2 above)
Comment (leopetter) [reply examples/rabbitmq/rabbitmq_example.go]: Fixed

File: rabbitmq_test.go (hunk 4 above)
Comment (leopetter) [reply rabbitmq_test.go]: Fixed

File: rabbitmq_test.go (hunk 5 above)
Comment (leopetter) [reply rabbitmq_test.go]: Fixed

**Pull request merged**